scripts/build.py

```diff
@@ -648,7 +724,19 @@ def generate_build_directory():
     save_hashes_as_json(HASHES_JSON, hashes)
 
     # Minify all template files copy them into build/templates/head.
-    build_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+    if not os.path.isdir(TEMPLATES_STAGING_DIR):
+        print 'Creating new %s folder' % TEMPLATES_STAGING_DIR
+        build_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+    else:
+        # Always rebuild HTML files.
+        build_HTML_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+        new_files_list = get_new_files_from_directory(
```

Comment:
In var names and function names, be careful to distinguish between files and filenames.

Reply:
Yup I was confusing myself with files and filenames. I have changed these `list of files` to `list of filenames`